sourcegraph/learn
# Remove package dependencies for ToC render

### What should this PR do?
Part of the work on [DEVED-93](https://sourcegraph.atlassian.net/browse/DEVED-93), this ticket, [DEVED-109](https://sourcegraph.atlassian.net/browse/DEVED-109), enables us to render our ToC without using external packages like `mdast-util-to-hast` and `mdast-util-toc`.

The hope here is to avoid future dependency complications from external packages that we can likely do without.

### Why are we making this change?
As part of the larger project in [DEVED-93](https://sourcegraph.atlassian.net/browse/DEVED-93), this PR asks the larger question of what we can do "in house", to avoid future dependency headaches. It rolls a version of the ToC using util methods rather than external packages.

### What are the acceptance criteria? 
- The ToC renders as expected, and styles look reasonable.

### How should this PR be tested?
- Check out this branch, nuke `node_modules`, and install dependencies with `npm ci`.
- Check out the ToC on records that have listed steps (`How To Install Sourcegraph on Windows with Docker`), nested steps (`How To Install Sourcegraph on Linux with Docker`), and no listed steps (`How To Search with Sourcegraph using Regular Expression Patterns`). Styles should look reasonable.
- Check functionality of ToC item links. 

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
